pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1827,15 +1827,57 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 		}
 
 		// Set VM CPU features
+		existingFeatures := make(map[string]struct{})
 		if vmi.Spec.Domain.CPU.Features != nil {
 			for _, feature := range vmi.Spec.Domain.CPU.Features {
+				existingFeatures[feature.Name] = struct{}{}
 				domain.Spec.CPU.Features = append(domain.Spec.CPU.Features, api.CPUFeature{
 					Name:   feature.Name,
 					Policy: feature.Policy,
 				})
 			}
 		}
 
+		/*
+						Libvirt validation fails when a CPU model is usable
+						by QEMU but lacks features listed in
+						`/usr/share/libvirt/cpu_map/[CPU Model].xml` on a node
+
+						To avoid the validation error mentioned above we will disable
+						some of the features in the `/usr/share/libvirt/cpu_map/[CPU Model].xml` files
+						for some of the cpu models.
+						Examples of validation error:
+			    		https://bugzilla.redhat.com/show_bug.cgi?id=2122283
+						https://gitlab.com/libvirt/libvirt/-/issues/304
+
+						This is temporary solution pending a permanent fix from Libvirt.
+		*/
+		cpuModelToFeaturesToDisable := map[string][]string{
+			"Opteron_G2":                {"svm"},
+			"Cascadelake-Server-noTSX":  {"mpx"},
+			"Cascadelake-Server":        {"mpx"},
+			"Icelake-Client-noTSX":      {"mpx"},
+			"Icelake-Client":            {"mpx"},
+			"Icelake-Server-noTSX":      {"mpx"},
+			"Icelake-Server":            {"mpx"},
+			"Skylake-Client-IBRS":       {"mpx"},
+			"Skylake-Client-noTSX-IBRS": {"mpx"},
+			"Skylake-Client":            {"mpx"},
+			"Skylake-Server-IBRS":       {"mpx"},
+			"Skylake-Server-noTSX-IBRS": {"mpx"},
+			"Skylake-Server":            {"mpx"},
+		}
```

Comment:
I would prefer to keep this knowledge outside of the virt-launcher. I think that this is in the cluster admin's domain to be able to regulate this. Even if this is a temporary workaround, I'd still prefer to let the admin adjust this. 
We could keep a sane default in virt-config and ultimately allow the admin to set/enable/adjust/disable this via kubevirt CR.

Reply:
@xpivarc what do you think?

Reply:
In addition to this list, advice from the libvirt devs regarding VM creation: On every VM we create, we should mandate "mpx=off"

As for enforcing the omission of flags at the virt-launcher level, I'm conflicted. This is a special case. mpx will *never* work under any circumstances on any cluster ever. If an admin were to attempt to forcibly enable mpx, that's just asking for trouble. Thus in this case I'm OK with the hard coded list, because making an admin consciously think on this is just noise.

Reply:
> I would prefer to keep this knowledge outside of the virt-launcher. I think that this is in the cluster admin's domain to be able to regulate this. Even if this is a temporary workaround, I'd still prefer to let the admin adjust this. We could keep a sane default in virt-config and ultimately allow the admin to set/enable/adjust/disable this via kubevirt CR.

While exposing this map through an API offers a quick solution to users without the need for back-porting and releasing, I have reservations about whether this is the appropriate fix. Given that it's a workaround, we should be cautious about further complicating the KubevirtCR API, which is already complex due to its many configuration options.

I propose allowing extensions to this map by enabling the addition of extra key-value pairs through annotations. This approach seems more logical, considering that the current solution is temporary and our strategy may change.

Reply:
> In addition to this list, advice from the libvirt devs regarding VM creation: On every VM we create, we should mandate "mpx=off"
> 
> As for enforcing the omission of flags at the virt-launcher level, I'm conflicted. This is a special case. mpx will _never_ work under any circumstances on any cluster ever. If an admin were to attempt to forcibly enable mpx, that's just asking for trouble. Thus in this case I'm OK with the hard coded list, because making an admin consciously think on this is just noise.


With the new changes, we will only enable these features if they are specifically requested in the VMI setup. For example, if ‘mpx’ is required, it will be enabled. This is safe because the launcher will include a node selector for the ‘mpx’ feature.